vernac/attributes.ml

```diff
@@ -11,13 +11,29 @@
 open CErrors
 
 (** The type of parsing attribute data *)
+type vernac_flag_type =
+  | FlagIdent of string
+  | FlagString of string
+
 type vernac_flags = vernac_flag list
 and vernac_flag = string * vernac_flag_value
 and vernac_flag_value =
   | VernacFlagEmpty
-  | VernacFlagLeaf of string
+  | VernacFlagLeaf of vernac_flag_type
   | VernacFlagList of vernac_flags
 
+let pr_vernac_flag_leaf = function
+  | FlagIdent b -> Pp.str b
+  | FlagString s -> Pp.(quote (str s))
+
+let rec pr_vernac_flag_value = let open Pp in function
+  | VernacFlagEmpty -> mt ()
+  | VernacFlagLeaf l -> str "=" ++ pr_vernac_flag_leaf l
+  | VernacFlagList s -> surround (pr_enum pr_vernac_flag s)
```

Comment:
Indeed, will amend, thanks! Tests for printing code are a bit tricky for a few reasons I think:

- we don't have the infrastructure for vernac, other than beautify
- beautify indeed does print some expressions wrong, so that's require too many changes at least in the scope of this PR
- we could use unit tests for that, but that seems a bit cumbersome

So I'd suggest we handle tests of the printing code as their own issue, as it seems important but not fully trivial.

That being said, if you find a quick way to do a test that would print the attributes of course I'll add it.

Reply:
Printing all the new error messages should be enough.